app/bundles/FormBundle/Event/FormFieldEvent.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * @copyright   2020 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\FormBundle\Event;
+
+use Mautic\CoreBundle\Event\CommonEvent;
+use Mautic\FormBundle\Entity\Field;
+
+class FormFieldEvent extends CommonEvent
```

Comment:
@kuzmany is there any plan with this event? I don't see it used in any subscriber.

Reply:
Yes, I wanted to use and play with delete children fields. But then not used it. But I decide let it there, because was already written and maybe some  devs could use it in future

Reply:
I've also try use foreign keys for parent_id to id but there was issue with cascade processing during the processing form (add/edit/delete). I was not able to figure out.

Reply:
That's what I thought but wanted to make sure I haven't overlooked anything. I'll update it a little bit with some defensive programming techniques since it's a new class so we have an opportunity to create it the right way and don't worry about BC breaks.

Reply:
Don't worry. We have branch for M2 and all your changes I will cherry pick there except M3 stuff. Thanks

Reply:
No need. The changes I did so far are compatible with M2 and M3. No need for cherry-picking. Once we are ready with bug fixes and complete the features then we can take this branch, clone it and make the final M3 touches there.

Reply:
👍